### DISCUSSION TITLE
force overwhite branch in remote if it exists and has diverged

### DISCUSSION THREAD

**Discussion by flavioop** [community/community#68815] [New to GitHub]
I was coding and when committing I selected the setting "force overwhite branch in remote if it exists and has diverged", I lost my commit history, I would like to recover it, how do I do it? It's possible?

**Comment by vishalkrishna8** [comment]
Hi @flavioop 

Here's a general guideline to attempt recovery:
* Check the reflog with `git reflog <branch-name>` to find the lost commit hash.

* Try to reset the branch to that commit with `git reset --hard <commit-hash>.`

* Check with collaborators for a backup.

* Check if your repository platform retains history.

* In the future, make regular backups to avoid data loss.

**Reply by flavioop** [reply]
Goodnight friend,
Thank you very much for your attention!
In fact, I have the codes saved, what I lost was the commit history and the codes in git that I use as my learning curriculum.
You can recover these commits that you were making daily, as you separated them by class.
Now I just have the application ready.

**Reply by vishalkrishna8** [reply]
If you found this helpful, you can mark my reply as answer.
